src/v/kafka/server/tests/partition_reassignments_test.cc

```diff
@@ -224,6 +224,33 @@ FIXTURE_TEST(
         }
     }
 
+    {
+        test_log.info("Unknown topic");
```

Comment:
Can we add a test for invalid replication factor

Reply:
This is the test for invalid replication factor, but instead of returning `invalid_replication_factor` like I was doing previously, now we are returning unknown topic because an undefined `topic_metadata` semantically means that the broker thinks the topic does not exist.

I thought we discussed this in the sync yesterday https://github.com/redpanda-data/redpanda/pull/8849#discussion_r1114502365

Reply:
Sync'ed offline

Reply:
> Can we add a test for invalid replication factor

This is in the ducktape tests.